common/pkg/libartifact/store/artifact.go

```diff
@@ -55,16 +56,16 @@ func (a *Artifact) GetDigest() (*digest.Digest, error) {
 
 type ArtifactList []*Artifact
 
-// GetByNameOrDigest returns an artifact, if present, by a given name
+// getByNameOrDigest returns an artifact, if present, by a given name
 // Returns an error if not found.
-func (al ArtifactList) GetByNameOrDigest(nameOrDigest string) (*Artifact, bool, error) {
+func (al ArtifactList) getByNameOrDigest(nameOrDigest string) (*Artifact, bool, error) {
```

Comment:
I see there is a lot of iteration over lists. In extreme cases, when the store has a lot of artifacts, this will cost us some time. I think there could be an optimization of this search.